src/client/Microsoft.Identity.Client/Platforms/Features/SLC/KeyMaterialManager.cs

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+
+namespace Microsoft.Identity.Client.Platforms.netcore
+{
+    /// <summary>
+    /// Provides X509_2 certificates and cryptographic key information for a Credential based 
+    /// Managed Identity-supported Azure resource. 
+    /// This class handles the retrieval or creation of X.509_2 certificates for authentication purposes,
+    /// including the determination of the cryptographic key type.
+    /// For more details, see https://aka.ms/msal-net-managed-identity.
+    /// </summary>
+    internal class ManagedIdentityCertificateProvider : IKeyMaterialManager
+    {
+        // The name of the key guard isolation property
+        private const string IsKeyGuardEnabledProperty = "Virtual Iso";
+
+        // Field to store the current crypto key type
+        private static CryptoKeyType s_cryptoKeyType = CryptoKeyType.None;
+
+        // Constants specifying the names for the key storage provider and key names
```

Comment:
Is there a documented source for these constants? Should it be added in the comments?

Reply:
added comments to enum